python/perspective/perspective/tests/table/test_update_arrow.py

```diff
@@ -667,6 +839,88 @@ def test_update_arrow_updates_dictionary_stream(self, util):
             "b": ["x", "y", None, "z"]
         }
 
+    def test_update_arrow_partial_updates_dictionary_stream(self, util):
+        data = [
+            ([0, 1, 1, None], ["a", "b"]),
+            ([0, 1, None, 2], ["x", "y", "z"])
+        ]
+        arrow_data = util.make_dictionary_arrow(["a", "b"], data)
+
+        tbl = Table({
+            "a": str,
+            "b": str
+        }, index="a")
+
+        tbl.update(arrow_data)
+
+        assert tbl.size() == 3
+        assert tbl.view().to_dict() == {
+            "a": [None, "a", "b"],
+            "b": ["z", "x", "y"]
+        }
+
+    @mark.skip
+    def test_update_arrow_partial_updates_dictionary_stream_duplicates(self, util):
```

Comment:
Painful issue.  As discussed, we can detect this if we care to sacrifice a bit of memory for a HashSet to detect collisions, but I'm far from convinced we want to.  I am curious if a dictionary encoded column in arrow is even considered valid?